modules/passkey/test/userStories/rotatePasskeyOwner.ts

```diff
@@ -0,0 +1,181 @@
+      user,
+      proxyFactory,
+      safeModuleSetup,
+      module,
+      entryPoint,
+      singleton,
+      signerFactory,
+      navigator,
+      verifier,
+      SafeL2,
+      signer,
+      credential,
+      safeAddress,
+    }
+  })
+
+  it('should execute a userOp with replaced WebAuthn signer as Safe owner', async () => {
+    // Step 1: Setup the contracts
+    const { user, module, entryPoint, verifier, navigator, SafeL2, signer, signerFactory, safeAddress } = await setupTests()
+
+    // Check the owners of the created Safe
+    const safeInstance = await ethers.getContractAt(SafeL2.abi, safeAddress)
+    expect(await safeInstance.getOwners()).to.deep.equal([user.address, signer])
+
+    // Step 2: Create a userOp to swap the passkey signer, sign it with EOA wallet
+
+    // Create a new WebAuthn credential for the new signer
+    const credentialNew = navigator.credentials.create({
+      publicKey: {
+        rp: {
+          name: 'Safe',
+          id: 'safe.global',
+        },
+        user: {
+          id: ethers.getBytes(ethers.id('chucknorris')),
+          name: 'chucknorris',
+          displayName: 'Chuck Norris',
+        },
+        challenge: ethers.toBeArray(Date.now()),
+        pubKeyCredParams: [{ type: 'public-key', alg: -7 }],
+      },
+    })
+
+    const publicKeyNew = decodePublicKey(credentialNew.response)
+    await (await signerFactory.createSigner(publicKeyNew.x, publicKeyNew.y, verifier.target)).wait()
+    const signerNew = await signerFactory.getSigner(publicKeyNew.x, publicKeyNew.y, verifier.target)
+
+    const interfaceSwapOwner = new ethers.Interface(['function swapOwner(address,address,address)'])
+    const data = interfaceSwapOwner.encodeFunctionData('swapOwner', [user.address, signer, signerNew])
+
+    // Build SafeOp
+    const safeOpSwapSigner = buildSafeUserOpTransaction(
+      safeAddress,
+      safeAddress,
+      0,
+      data,
+      await entryPoint.getNonce(safeAddress, 0),
+      await entryPoint.getAddress(),
+      false,
+      true,
+      {
+        verificationGasLimit: 700000,
+        callGasLimit: 2000000,
+        maxFeePerGas: 10000000000,
+        maxPriorityFeePerGas: 10000000000,
+      },
+    )
+
+    // EOA wallet signs the userOp to swap the Passkey signer
+    const signatureSwapSigner = buildSignatureBytes([
+      {
+        signer: signer as string,
+        data: buildSignatureBytes([await signSafeOp(user, await module.getAddress(), safeOpSwapSigner, await chainId())]),
+      },
+    ])
+
+    const packedUserOpSwapSigner = buildPackedUserOperationFromSafeUserOperation({
+      safeOp: safeOpSwapSigner,
+      signature: signatureSwapSigner,
+    })
+
+    // Step 3: Execute the userOp to swap the passkey signer
+
+    // Send 1 ETH to the Safe
+    await user.sendTransaction({ to: safeAddress, value: ethers.parseEther('1') }).then((tx) => tx.wait())
+
+    await (await entryPoint.handleOps([packedUserOpSwapSigner], user.address)).wait()
```

Comment:
```suggestion
    await user.sendTransaction({ to: safeAddress, value: ethers.parseEther('1') })

    await entryPoint.handleOps([packedUserOpSwapSigner], user.address)
```

Reply:
Agree. `.wait()`-ing is not needed on `hardhat` network.